addons/account/models/partner.py

```diff
@@ -172,6 +172,9 @@ def get_fiscal_position(self, partner_id, delivery_id=None):
         # if no delivery use invoicing
         if delivery_id:
             delivery = PartnerObj.browse(delivery_id)
+            # If partner and delivery have the same vat prefix, use invoicing
+            if delivery.vat and partner.vat and delivery.vat[:2] == partner.vat[:2]:
+                delivery = partner
```

Comment:
This seems more readable to me, tell me what you think about it

```python
partner = self.env['res.partner'].browse(partner_id)
delivery = self.env['res.partner'].browse(delivery_id)

# If partner and delivery have the same vat prefix, use invoicing
if not (delivery.vat and partner.vat and delivery.vat[:2] == partner.vat[:2]):
    delivery = partner
```

Reply:
Indeed, it looks better :+1: 
I didn't know the `browse()` method could handle a `None` value without throwing an error :ok: